profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/LinuxStackFramesCollector.cpp

```diff
@@ -253,54 +253,22 @@ std::int32_t LinuxStackFramesCollector::CollectCallStackCurrentThread()
 {
     try
     {
-        std::int32_t resultErrorCode;
+        // Collect data for TraceContext tracking:
+        bool traceContextDataCollected = TryApplyTraceContextDataFromCurrentCollectionThreadToSnapshot();
 
-        {
-            // Collect data for TraceContext tracking:
-            bool traceContextDataCollected = TryApplyTraceContextDataFromCurrentCollectionThreadToSnapshot();
-
-            // Now walk the stack:
-
-            unw_context_t uc;
-            unw_getcontext(&uc);
-
-            unw_cursor_t cursor;
-            unw_init_local(&cursor, &uc);
-
-            // After every lib call that touches non-local state, check if the StackSamplerLoopManager requested this walk to abort:
-            if (IsCurrentCollectionAbortRequested())
-            {
-                AddFakeFrame();
-                return E_ABORT;
-            }
-
-            resultErrorCode = unw_step(&cursor);
-
-            while (resultErrorCode > 0)
-            {
-                // After every lib call that touches non-local state, check if the StackSamplerLoopManager requested this walk to abort:
-                if (IsCurrentCollectionAbortRequested())
-                {
-                    AddFakeFrame();
-                    return E_ABORT;
-                }
+        // Now walk the stack:
+        auto [data, size] = Data();
 
-                unw_word_t nativeInstructionPointer;
-                resultErrorCode = unw_get_reg(&cursor, UNW_REG_IP, &nativeInstructionPointer);
-                if (resultErrorCode != 0)
-                {
-                    return resultErrorCode;
-                }
+        auto count = unw_backtrace((void**)data, size);
 
-                if (!AddFrame(nativeInstructionPointer))
-                {
-                    return S_FALSE;
-                }
-
-                resultErrorCode = unw_step(&cursor);
-            }
+        if (count == 0)
+        {
+            AddFakeFrame();
```

Comment:
if we had a fake frame, the number of frames will be 1 and in the stacksampler loop  (for example), we will process this `empty` callstack. (the return value `S_FALSE` and al is kept only for statistic. We process a sample if the callstack is not empty)

maybe we can just skip adding a fake frame in that case, no?

Reply:
Right. I was mimicking the "collection aborted" case of the previous code, but I guess it's closer to the "'error" case, where didn't add fake frames. I'll fix this.